pkg/apis/core/validation/validation.go

```diff
@@ -1020,7 +1019,11 @@ func validateDownwardAPIVolumeFile(file *core.DownwardAPIVolumeFile, fldPath *fi
 			allErrs = append(allErrs, field.Invalid(fldPath, "resource", "fieldRef and resourceFieldRef can not be specified simultaneously"))
 		}
 	} else if file.ResourceFieldRef != nil {
-		allErrs = append(allErrs, validateContainerResourceFieldSelector(file.ResourceFieldRef, &validContainerResourceFieldPathExpressions, fldPath.Child("resourceFieldRef"), true)...)
+		localValidContainerResourceFieldPathPrefixes := sets.NewString()
+		if opts.AllowDownwardAPIHugePages {
+			localValidContainerResourceFieldPathPrefixes = localValidContainerResourceFieldPathPrefixes.Union(validContainerResourceFieldPathPrefixes)
+		}
```

Comment:
nit, precompute the "with hugepages" and "without hugepages" sets as package vars and just select the appropriate one to use here... sets are expensive